src/librbd/ImageCtx.h

```diff
@@ -148,7 +148,11 @@ namespace librbd {
 
     file_layout_t layout;
 
-    cache::ImageCache *image_cache = nullptr;
+    cls::rbd::ImageCacheState image_cache_state;
```

Comment:
But we could add a single item here that contained everything about whatever ImageCache exists (or will exist, or used to exist) in this ImageCtx?

Reply:
Add a proposal so that I can see what you are thinking. I want to keep this cache isolated as much as possible instead of branching out and leaving little tendrils throughout the codebase.

Reply:
The image state is removed from ImageCtx.